components/library/glew/Makefile

```diff
@@ -0,0 +1,115 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov <jimklimov@gmail.com>
+#
+
+# NOTE: This project links with /usr/openwin/lib/libGL.so.1, which unrolls
+# to /usr/lib/GL/libGL.so.1 which is a symlink to run-time (hardware-related?)
+# mediator in the ../../../var/run/opengl/lib/libGL.so.1. So far so good...
+# In my case (GZ on a VirtualBox host, even though it is dual-booted and runs
+# OI/Hipster natively at the moment) this library symlink points back to
+# /usr/lib/mesa/libGL.so.1 and this one object points to /usr/lib/VBoxOGL.so
+# rather than the nearby mesa objects. This breaks the build because pkgmogrify
+# insists on objects coming from known repo's, and because ultimate objects do
+# refer to virtualbox rather than libs expected available on all deployments!
+# Perhaps we should *require* do the builds in a dedicated local zone?..
+# TODO: Find out the root-cause and attach a patch-file with a fix to this.
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		glew
+COMPONENT_VERSION=	1.13.0
+COMPONENT_SUMMARY=	The OpenGL Extension Wrangler Library
+COMPONENT_CLASSIFICATION=System/Libraries
+COMPONENT_FMRI=		library/$(COMPONENT_NAME)
+COMPONENT_LICENSE=	GLEW,Mesa3D,Khronos
+COMPONENT_LICENSE_FILE=	LICENSE.txt
+COMPONENT_PROJECT_URL=	http://$(COMPONENT_NAME).sourceforge.net/
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tgz
+COMPONENT_ARCHIVE_HASH= \
+	sha256:aa25dc48ed84b0b64b8d41cdd42c8f40f149c37fa2ffa39cd97f42c78d128bc7
+COMPONENT_ARCHIVE_URL= \
+	http://sourceforge.net/projects/$(COMPONENT_NAME)/files/$(COMPONENT_NAME)/$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+
+PATH=/usr/gnu/bin:/usr/bin
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/justmake.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+COMPONENT_PREP_ACTION = \
```

Comment:
A couple of questions here:
1) how do you do that? I'm not a linker alien yet =D
Any examples in our components worth flattering and paying tribute to?
2) given presence of ogl-select in overall architecture, is mesa the only
valid variant, or one bound to be present even if e.g. hardware-specific
ones are given, or otherwise a lowest-common denominator we can safely link
against (and work anywhere, and preferably not only with mesa but
accelerated by the optimal driver/lib chosen via ogl-select at the end-user
machine)?

Thanks, Jim

--
Typos courtesy of GMail on my Samsung Android
07.03.2016 18:34 пользователь "Aurélien Larcher" <notifications@github.com>
написал:

> In components/library/glew/Makefile
> <https://github.com/OpenIndiana/oi-userland/pull/1689#discussion_r55239206>
> :
>
> > +COMPONENT_LICENSE_FILE=	LICENSE.txt
> > +COMPONENT_PROJECT_URL=	http://$(COMPONENT_NAME).sourceforge.net/
> > +COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
> > +COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tgz
> > +COMPONENT_ARCHIVE_HASH= \
> > +	sha256:aa25dc48ed84b0b64b8d41cdd42c8f40f149c37fa2ffa39cd97f42c78d128bc7
> > +COMPONENT_ARCHIVE_URL= \
> > +	http://sourceforge.net/projects/$(COMPONENT_NAME)/files/$(COMPONENT_NAME)/$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
> > +
> > +PATH=/usr/gnu/bin:/usr/bin
> > +
> > +include $(WS_TOP)/make-rules/prep.mk
> > +include $(WS_TOP)/make-rules/justmake.mk
> > +include $(WS_TOP)/make-rules/ips.mk
> > +
> > +COMPONENT_PREP_ACTION = \
>
> You should point to mesa libs and set rpath accordingly.
> The mistake is to rely on the default path which points to runtime libs.
> I fixed the header path in the mesa update, do not remember if I pushed
> it...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/1689/files#r55239206>.
>